users/views.py

```diff
@@ -179,6 +195,108 @@ def get(self, request, user_id, token):
                 status=status.HTTP_500_INTERNAL_SERVER_ERROR
             )
 
+class ResendEmailView(APIView):
+    """
+    API view to resend the email verification link.
+
+    This view allows users to request a new verification email to be sent to their
+    registered or pending email address. It supports optional updating of the pending email
+    and token generation. The response does not disclose whether the user exists
+    for security reasons.
+
+    Attributes:
+        permission_classes (list): List of permission classes allowing unrestricted access.
+        throttle_scope (str): Named scope for rate limiting resend email requests.
+        throttle_classes (list): List of throttling classes applied to the view.
+    """
+    permission_classes = [AllowAny]
+    throttle_scope = "resend_email"
+    throttle_classes = [ScopedRateThrottle]
+
+    def post(self, request):
+        """
+        Handle POST request to resend the email verification link.
+
+        Validates the input data, retrieves the user by user_id, updates the pending email if provided,
+        generates a new token if not supplied, constructs the verification URL, renders
+        email templates (HTML and plain text), sends the email, and returns a generic
+        success response regardless of whether the user exists.
+
+        Args:
+            request (Request): DRF request object containing user_id (required), 
+                optional email (new pending email), and optional token.
+
+        Returns:
+            Response: DRF Response object with HTTP 202 Accepted status and a message
+                indicating that if the account exists, a verification email has been sent.
+        """
+        serializer = ResendEmailSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        user_id = serializer.validated_data["user_id"]
+        email = serializer.validated_data.get("email")
+        token = serializer.validated_data.get("token")
+
+        try:
+            user = User.objects.get(user_id=user_id)
```

Comment:
Instead of returning immediately if user.is_active, consider still updating the pending_email if a new one is provided (depending on business logic).